software/litepcie-user-library/litepcie_util.c

```diff
@@ -418,6 +418,9 @@ static int dma_test(uint8_t zero_copy, uint8_t external_loopback, int data_width
     // We purposefully put this afterward, so we can visually see we passed `litepcie_dma_init()`.
     printf("---------------------\n");
 
+    // Disable the DMA synchronizer, as we want this loopback to be independent of anything else:
+    litepcie_writel(dma.fds.fd, CSR_PCIE_DMA0_SYNCHRONIZER_BYPASS_ADDR, 1);
```

Comment:
I think you will also want:
https://github.com/JuliaComputing/xtrx_julia/blob/70e8a9118b04eb12e33078b7a99f2569d7a4ec3c/software/soapysdr-xtrx/XTRXDevice.cpp#L116-L118

Reply:
Which should set: https://github.com/enjoy-digital/litepcie/blob/9e3cdc27ac1631a5e188bbbb026d86f78aaa3810/litepcie/frontend/dma.py#L546-L550